File: py/1242.py
def main():
    while True:
        try:
            rnaa = input()
            count = 0
            stack = []

            for i in rnaa:
                if (
                    (len(stack) > 0) and (
                        (i == 'B' and stack[-1] == 'S') or
                        (i == 'S' and stack[-1] == 'B') or
                        (i == 'C' and stack[-1] == 'F') or
                        (i == 'F' and stack[-1] == 'C')
                    )
                ):
                    count += 1
                    stack.pop(-1)
                else:
                    stack.append(i)

            print(count)

        except EOFError:
            break


main()

File: cpp/1242.cpp
#include <iostream>
#include <string>
#include <stack>

using namespace std;

int main()
{
    string rnaa;

    while (cin >> rnaa) {
        int count = 0, len = rnaa.length(), i;
        stack<char> rnaa_stack;

        for (i = 0; i < len; i++) {
            if (!rnaa_stack.empty() && (
                (rnaa[i] == 'B' && rnaa_stack.top() == 'S') ||
                (rnaa[i] == 'S' && rnaa_stack.top() == 'B') ||
                (rnaa[i] == 'C' && rnaa_stack.top() == 'F') ||
                (rnaa[i] == 'F' && rnaa_stack.top() == 'C')
            )) {
                count++;
                rnaa_stack.pop();
            } else {
                rnaa_stack.push(rnaa[i]);
            }
        }

        cout << count << '\n';        
    }

    return 0;
}